nipype/algorithms/confounds.py

```diff
@@ -249,9 +252,11 @@ class FramewiseDisplacement(BaseInterface):
     }]
 
     def _run_interface(self, runtime):
-        mpars = np.loadtxt(self.inputs.in_plots)  # mpars is N_t x 6
+        mpars = np.loadtxt(self.inputs.in_file)  # mpars is N_t x 6
         diff = mpars[:-1, :] - mpars[1:, :]
         diff[:, :3] *= self.inputs.radius
+        if self.inputs.parameter_source == "AFNI":
+            diff[:, :3] *= (np.pi / 180)
         fd_res = np.abs(diff).sum(axis=1)
```

Comment:
I'd suggest modifying the `mpars` rather than the diff, just to make it a little clearer what's going on. Following also adds FSFAST support.

```Python
mpars = np.loadtxt(self.inputs.in_file)  # mpars is N_t x 6 (N_t x 7 for FSFAST)
if self.inputs.parameter_source in ('AFNI', 'FSFAST'):
    mpars = mpars[np.asarray([4, 5, 3, 1, 2, 0]) + (mpars.shape[1] > 6)]
    mpars[:, 3:] *= np.pi / 180
diff = np.diff(mpars, axis=0)
diff[:, :3] *= self.inputs.radius
fd_res = np.abs(diff).sum(axis=1)